crates/saya/core/src/lib.rs

```diff
@@ -348,13 +371,15 @@ impl Saya {
         let expected_fact = poseidon_hash_many(&[program_hash, program_output_hash]).to_string();
         info!(target: LOG_TARGET, expected_fact, "Expected fact.");
 
+        sleep(Duration::from_millis(5000)).await;
```

Comment:
Wondering why this sleep is still required because the verify function is using correctly the `wait: true`. Would you mind adding a comment here to ensure this can be tested in the future to be removed? :)